### PR TITLE
[Method References 2] LPS-176936 Convert method reference to field reference in class LiferayResourceOwnerAccessTokenGrantHandler

### PR DESCRIPTION
**Pull request by W2jx**
resend

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/3916

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:oauth2

**Comment by Kvale1733** [comment]
@W2jx No regressions.

**Comment by W2jx** [comment]
no conflicts(4/3 9:31)

**Comment by W2jx** [comment]
no conflicts(4/4 10:27)

**Comment by W2jx** [comment]
no conflicts(4/4 10:27)

**Comment by W2jx** [comment]
no conflicts(4/7 16:28)

File: modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java [hunk 1]
```diff
@@ -105,39 +105,18 @@ protected boolean isGrantHandlerEnabled() {
...
-	)
-	protected void setLiferayOAuthDataProvider(
-		LiferayOAuthDataProvider liferayOAuthDataProvider) {
+	private static final Log _log = LogFactoryUtil.getLog(
+		LiferayResourceOwnerAccessTokenGrantHandler.class);
 
-		_liferayOAuthDataProvider = liferayOAuthDataProvider;
-	}
+	@Reference
+	private LiferayOAuthDataProvider _liferayOAuthDataProvider;
```
Comment (tinatian) [inline modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/endpoint/access/token/grant/handler/LiferayResourceOwnerAccessTokenGrantHandler.java:131]: @W2jx Before you change, this is marked as Dynamic too, why you remove it?

**Comment by tinatian** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/133047

**Pull request closed without merging**